documentation/docs/05-misc/02-typescript.md

```diff
@@ -2,6 +2,198 @@
+
+```ts
+/// file: svelte.config.js
+import { vitePreprocess } from '@sveltejs/vite-plugin-svelte';
+
+const config = {
+	preprocess: vitePreprocess()
+};
+
+export default config;
+```
+
+If you're using SvelteKit, `vitePreprocess` is imported from `@sveltejs/kit/vite`
+
+```ts
+/// file: svelte.config.js
+// @noErrors
+import { vitePreprocess } from '@sveltejs/kit/vite';
+
+const config = {
+	preprocess: vitePreprocess()
+};
+
+export default config;
+```
+
+## `<script lang="ts">`
+
+To add types to your JavaScript inside the `<script>` tag, you can use the `script` tag's `lang` attribute:
+
+```svelte
+<script lang="ts">
+	let name: string = 'world';
+
+	function greet(name: string) {
+		alert(`Hello, ${name}!`);
+	}
+</script>
+```
+
+This enables all TypeScript features inside your Svelte components.
+
+## Props
+
+Props can be directly typed by using the `export let` syntax:
+
+```svelte
+<script lang="ts">
+	export let name: string;
+</script>
+```
+
+These are ultimately just variable declarations, so it just works.
+
+> See this [guide](https://github.com/dummdidumm/rfcs/blob/ts-typedefs-within-svelte-components/text/ts-typing-props-slots-events.md#typing-props) for more information about the experimental `$$Props` interface.
+
+## Slots
+
+Slots and slot props types are inferred from the type of slot props passed to them.
+
+For example:
+
+```svelte
+<script lang="ts">
+	export let name: string;
+</script>
+
+<slot {name} />
+
+<!-- Later -->
+<Comp let:name>
+	<!--    ^ Inferred as string -->
+	{name}
+</Comp>
+```
+
+> See this [guide](https://github.com/dummdidumm/rfcs/blob/ts-typedefs-within-svelte-components/text/ts-typing-props-slots-events.md#typing-slots) for more information about the experimental `$$Slots` interface.
+
+## Events
+
+Events can be typed with `createEventDispatcher`:
+
+```svelte
+<script lang="ts">
+	import { createEventDispatcher } from 'svelte';
+
+	const dispatch = createEventDispatcher<{
+		click: string;
+		type: string;
+	}>();
+
+	function handleClick() {
+		dispatch('click', { detail: 'hello' });
+	}
+
+	function handleType() {
+		dispatch('type', { detail: 'world' });
+	}
+</script>
+
+<button on:click={handleClick} on:keydown={handleType}>Click</button>
+```
+
+However, sometimes an event dispatcher may be coming from another file/package. For example:
+
+```svelte
+<script>
+	import { mixinDispatch } from 'somewhere';
+
+	function handleClick() {
+		mixinDispatch.mixinEvent('foo');
+	}
+</script>
+
+<button on:click={handleClick}>Mixin</button>
+```
+
+In this case, svelte can't understand the custom event types. So, you can use the experimental `$$Events` inerface.
+
+> See this [guide](https://github.com/dummdidumm/rfcs/blob/ts-typedefs-within-svelte-components/text/ts-typing-props-slots-events.md#typing-events) for more information about the experimental `$$Events` interface.
+
+## Limitations
+
+### No TS in Template
```

Comment:
I think we've generally not used capitalization in titles, so probably more consistent to use lowercase here
```suggestion
### No TS in template
```